backend/inspirehep/cds/api.py

```diff
@@ -0,0 +1,208 @@
+def sync_identifiers(since=None):
+    LOGGER.info("Starting CDS Sync", since=since)
+    if not current_app.config.get("FEATURE_FLAG_ENABLE_CDS_SYNC", False):
+        LOGGER.warning("CDS sync is currently disabled by feature flag.")
+        return
+
+    last_run = CDSRun.get_last_successful_run()
+    last_run_date = last_run.date if last_run else None
+
+    since = since or last_run_date
+    if not since:
+        LOGGER.error(
+            "CDS Sync failed. No `since` provided and no successful runs in DB. Aborting."
+        )
+        raise CDSSyncError("Missing `since` date")
+
+    task_id = CDSRun.new_run()
+    db.session.commit()
+    try:
+        run_sync(since)
+    except Exception as err:
+        LOGGER.exception("CDS Sync failed.", error=err)
+        CDSRun.update_status(task_id, status=CDSRunStatus.ERROR, message=str(err))
+        db.session.commit()
+        raise
+    else:
+        LOGGER.info("CDS Sync finished successfully.")
+        CDSRun.update_status(task_id, status=CDSRunStatus.FINISHED)
+        db.session.commit()
+
+
+def run_sync(since):
+    """Starts sync
+    Args:
+        since(datetime.date): since when records should be synced
+    """
+    LOGGER.info("Starting CDS identifiers sync", since=since)
+    cds_api = current_app.config.get("CDS_SERVER_API")
+    service = "inspire2cdsids"
+    if not cds_api:
+        raise MissingCDSServerConfig("CDS Server config is missing.")
+    url = parse.urljoin(cds_api, service)
+    response = requests.get(url, params={"since": since})
+    response.raise_for_status()
+    process_cds_response(response.json())
+
+
+def process_cds_response(response):
+    try:
+        for idx, record in enumerate(response.get("hits", [])):
+            process_cds_record(record)
+            if idx % CHUNK_SIZE:
+                db.session.commit()
+        db.session.commit()
+
+    except SQLAlchemyError as err:
+        LOGGER.exception("Cannot commit a batch of updates from CDS.", error=err)
+        raise
+
+
+def get_record_for_pid_or_none(
+    pid_type,
+    pid_value,
+):
+    try:
+        record_object = LiteratureRecord.get_record_by_pid_value(
+            pid_value=pid_value, pid_type=pid_type
+        )
+        LOGGER.info(
+            "Found record based on CDS ids", pid_type=pid_type, pid_value=pid_value
+        )
+    except PIDDoesNotExistError:
+        return None
+    else:
+        return record_object
+
+
+def query_report_number(report_number):
+    query = Q("match", report_numbers__value__fuzzy=report_number)
+    source = ["control_number"]
+    results = LiteratureSearch().query(query).source(source).execute()
+    if len(results.hits) == 1:
+        control_number = results.hits[0]["control_number"]
+        return get_record_for_pid_or_none(
+            "lit",
+            control_number,
+        )
+    return None
+
+
+def get_record_for_provided_ids(control_numbers, arxivs, dois, report_numbers):
+    for control_number in control_numbers:
```

Comment:
I would split the logic for it's id for this function, currently is too long

Reply:
It's shorter than `process_cds_record` function but as you prefer.